core/src/cdn_url.rs

```diff
@@ -16,6 +16,8 @@ use protocol::storage_resolve::StorageResolveResponse as CdnUrlMessage;
 #[derive(Debug, Clone)]
 pub struct MaybeExpiringUrl(pub String, pub Option<Date>);
 
+const CDN_URL_EXPIRY_MARGIN_SECONDS: i64 = 5 * 60;
```

Comment:
I ended up using `time::duration` rather than `std::time::duration` after losing all patience with the incompatible i64 and u64 types in each. Maybe there's a simple way and I am being daft, please suggest if so!

Reply:
I think that's fine, seeing how that `Date` is also a `time` thing.